tasks/Module 5/task_2.py

```diff
@@ -0,0 +1,233 @@
+        initial_position = hero.position
+        # print('initial position', initial_position)
+        if direction in ("north", "u"):
+            # to not cross border
+            aligned_init_pos = (
+                len(self.field) if initial_position[0] == 0 else initial_position[0]
+            )
+            # get current contenf of the cell
+            cell_content = self.field[aligned_init_pos - 1][initial_position[1]]
+            # move hero to new position
+            self.field[aligned_init_pos - 1][initial_position[1]] = HERO_SYMBOL
+            # change hepo position on hero object
+            hero.position = (aligned_init_pos - 1, initial_position[1])
+        elif direction in ("south", "d"):
+            aligned_init_pos = (
+                -1
+                if initial_position[0] == len(self.field) - 1
+                else initial_position[0]
+            )
+            cell_content = self.field[aligned_init_pos + 1][initial_position[1]]
+            self.field[aligned_init_pos + 1][initial_position[1]] = HERO_SYMBOL
+            hero.position = (aligned_init_pos + 1, initial_position[1])
+        elif direction in ("west", "l"):
+            aligned_init_pos = (
+                len(self.field[0]) if initial_position[1] == 0 else initial_position[1]
+            )
+            cell_content = self.field[initial_position[0]][aligned_init_pos - 1]
+            self.field[initial_position[0]][aligned_init_pos - 1] = HERO_SYMBOL
+            hero.position = (initial_position[0], aligned_init_pos - 1)
+        elif direction in ("east", "r"):
+            aligned_init_pos = (
+                -1
+                if initial_position[1] == len(self.field[0]) - 1
+                else initial_position[1]
+            )
+            cell_content = self.field[initial_position[0]][aligned_init_pos + 1]
+            self.field[initial_position[0]][aligned_init_pos + 1] = HERO_SYMBOL
+            hero.position = (initial_position[0], aligned_init_pos + 1)
+        # clear initial hero position
+        self.field[initial_position[0]][initial_position[1]] = " "
+        # print('position after move', hero.position)
+        return cell_content
+
+    def __str__(self):
+        """Draw field."""
+        to_print = ""
+        for _, row in enumerate(self.field):
+            to_print += "+---" * self.columns + "+\n"
+            to_print += (
+                "| "
+                + " | ".join(map(lambda x: ITEM_SYMBOL if isinstance(x, Item) else x, row))
+                + " |\n"
+            )
+        to_print += "+---" * self.columns + "+\n"
+        return to_print
+
+
+class Game:
+    """Main Game class."""
+
+    def __init__(self, columns, rows, hero):
+        """Init Game instance."""
+        self.field = Board(columns, rows)
+        self.field.place_items()
+        self.hero = hero
+        self.field.set_hero(self.hero)
+        self.debug = False
+
+    def __handle_responce(self, responce):
```

Comment:
Why double underscore here?

Reply:
like private methods

Reply:
Probably it is not as private as you would like it to be. You can still access this method as:
```python
game = Game()
game._Game__handle_response(response)
```
Actually Python does not have a concept of private method like other OO languages. Usually people use single underscore (eg. _handle_response) to indicate that method is private, and user of this class shouldn't use it. Of course nothing prevents him from using it anyway, maybe some linters complain about it.

When you start method name with double underscore it just adds `_ClassName` at the beginning of method name. It is called "name mangling" and its purpose it to prevent subclasses from overriding this. You can read more about it here: https://docs.python.org/3/tutorial/classes.html#private-variables